src/deadline/client/config/config_file.py

```diff
@@ -247,8 +247,11 @@ def _reset_directory_permissions_windows(directory: Path, username: str, permiss
             *_get_grant_args(username, permissions),
             # On Windows, both SYSTEM and the Administrators group normally
             # have Full Access to files in the user's home directory.
-            *_get_grant_args("Administrators", permissions),
-            *_get_grant_args("SYSTEM", permissions),
+            # Use SIDs to represent the Administrators and SYSTEM to
```

Comment:
Did you happen to test out the code on lines 227-228 as well for locale issues? They refer to Administrator & SYSTEM by name.
```
            if (
                ad_user != username
                and principal != "BUILTIN\\Administrators"
                and principal != "NT AUTHORITY\\SYSTEM"
            ):
                principals_to_remove.append(ad_user)
```

Reply:
Would testing a submission not be enough? I looked at changing this but did not see a way to return the SID from icacls. What would happens is that the BUILTIN/SYSTEM would be removed, however we are adding them again in the following grant. Not sure what the intent of this code is.

Reply:
Just as a quick example:
```
PS C:\Users\Administrator\.deadline> icacls C:\Users\Administrator\.deadline\
C:\Users\Administrator\.deadline\ NT AUTHORITY\SYSTEM:(OI)(CI)(F)
                                  BUILTIN\Administrators:(OI)(CI)(F)
                                  EC2AMAZ-E0K168U\Administrator:(OI)(CI)(F)
                                  EC2AMAZ-E0K168U\Administrator:(I)(OI)(CI)(F)

Successfully processed 1 files; Failed processing 0 files
PS C:\Users\Administrator\.deadline> icacls C:\Users\Administrator\.deadline\ /remove Administrators
processed file: C:\Users\Administrator\.deadline\
Successfully processed 1 files; Failed processing 0 files
PS C:\Users\Administrator\.deadline> icacls C:\Users\Administrator\.deadline\ /remove SYSTEM
processed file: C:\Users\Administrator\.deadline\
Successfully processed 1 files; Failed processing 0 files
PS C:\Users\Administrator\.deadline> icacls C:\Users\Administrator\.deadline\
C:\Users\Administrator\.deadline\ EC2AMAZ-E0K168U\Administrator:(OI)(CI)(F)
                                  EC2AMAZ-E0K168U\Administrator:(I)(OI)(CI)(F)

Successfully processed 1 files; Failed processing 0 files
PS C:\Users\Administrator\.deadline> icacls C:\Users\Administrator\.deadline /grant 'Administrador:(OI)(CI)(F)' /T /grant '*S-1-5-32-544:(OI)(CI)(F)' /T /grant '*S-1-5-18:(OI)(CI)(F)' /T
Successfully processed 71 files; Failed processing 0 files
PS C:\Users\Administrator\.deadline> icacls C:\Users\Administrator\.deadline\
C:\Users\Administrator\.deadline\ NT AUTHORITY\SYSTEM:(OI)(CI)(F)
                                  BUILTIN\Administrators:(OI)(CI)(F)
                                  EC2AMAZ-E0K168U\Administrator:(OI)(CI)(F)
                                  EC2AMAZ-E0K168U\Administrator:(I)(OI)(CI)(F)

Successfully processed 1 files; Failed processing 0 files
PS C:\Users\Administrator\.deadline>
```

Reply:
Yeah, I think that we're okay here. As you say, it'll just remove and then re-add the same things. It's probably an optimization.